upgrade castle integration

bartes opened the pull request:
**What does this PR do?**
- switched castle integration to the new castle script (https://www.npmjs.com/package/@castleio/castle-js).  old one (https://www.npmjs.com/package/castle.js) is no longer maintained. Maps segment.page to https://docs.castle.io/docs/sdk-browser#castlepage and segment.track to https://docs.castle.io/docs/sdk-browser#castlecustom .

**Are there breaking changes in this PR?**
 - no , legacy integrations should work without any changes

**Testing**
-  test suite has been updated and extended accordingly

**Any background context you want to provide?**
- this PR switches setup to the new script which includes additional features

**Is there parity with the server-side/android/iOS integration components (if applicable)?**
- not yet 

**Does this require a new integration setting? If so, please explain how the new setting works**
- no (but some legacy options are no longer needed, and were removed)

**Links to helpful docs and other external resources**
- https://docs.castle.io/docs/sdk-browser

andreloureiro88 commented on integrations/castle/lib/index.js:
```diff
@@ -47,57 +51,138 @@ Castle.prototype.initialize = function() {
+    objCase.del(traits, 'phone');
+  }
+
+  if (objCase.find(traits, 'registeredAt')) {
```
maybe we should map this consts somewhere and reuse them

JimGochee2 reviewed (approved):
I'm good with the description in the PR.

gs-nwolfe committed:
Aptrinsic/region (#673)

* Aptrinsic (Gainsight PX) - Add region support
WIP

* Aptrinsic (Gainsight PX) - Add region support
Added tests for region setting

* Aptrinsic (Gainsight PX) - Add region support
Changed new option name to be more accurate

* Aptrinsic (Gainsight PX) - Add support for new US2 datacenter

* Aptrinsic (Gainsight PX) - Add support for new US2 datacenter
Version bump in history and package.json

bartes committed:
added user address

komailo committed:
Remove Circle CI badge from README.md CI-43 (#670)

bnorton committed:
Chameleon: Add config for fastUrl and pass integration-specific options to identify. (#676)

* Add config for fastUrl and pass integration-specific options to identify.

* Update version.

chrisradek committed:
Added support for Hotjar Events API. (#660) (#677)

* Added support for Hotjar Events API.

* Start passing properties() to the Event API

* Updated tests to check track with/without properties

Co-authored-by: julio.alegria <julio.alegria@hotjar.com>
Co-authored-by: Ariel Silvestri <arielsilvestri0715@gmail.com>

Co-authored-by: Julio M. Alegria <juliomalegria@users.noreply.github.com>
Co-authored-by: julio.alegria <julio.alegria@hotjar.com>
Co-authored-by: Ariel Silvestri <arielsilvestri0715@gmail.com>

bnorton committed:
Incorporate changes for fastUrl. Update version to 2.0.3 (#684)

pooyaj committed:
Update HISTORY.md

pooyaj committed:
Pj/tester (#688)

* adding the build  command

* adding the webpack config for the destinations

* adding tester app

* yarn.lock

pooyaj committed:
Adding Tester Docs + Removing the old Compiler  (#691)

* update readme

* remove the legacy compiler

* readme tweak

rhall-twilio committed:
add support for snake_case for contentAssetId in Adobe Heartbeat (#692)

* add support for snake_case for contentAssetId in Adobe Heartbeat

* prefer camelCase over snake_case instead

joshuarrrr committed:
update Parse.ly tracker to  config-free tracker (#417)

Co-authored-by: Keith Bourgoin <keith.bourgoin@gmail.com>

pooyaj committed:
Update parsely version

pooyaj committed:
fixing source-maps (#694)

pooyaj committed:
fix variable name typo (#696)

bartes committed:
merge

bartes committed:
update history

pooyaj committed:
Adding regional api support for intercom (#697)

* adding regional api support for intercom

* bump version

chrisradek committed:
update all @segment/trample to same minimum version (#699)

bartes committed:
Merge remote-tracking branch 'upstream/master' into castle_v3
